File: data/compilers_comparation/src/examples/second_example/src/func7.c
#include "second_example.h" 
array_t* func7(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   unsigned int loop65 = 0;
   unsigned int loopLimit65 = (rand()%loopsFactor)/2 + 1;
   for(; loop65 < loopLimit65; loop65++) {
      if(PATH0 & 1) {
         array_t* array224;
         if (pCounter > 0) {
            array224 = vars->data[--pCounter];
            array224->refC++;
            DEBUG_COPY(array224->id);
         } else {
            array224 = (array_t*)malloc(sizeof(array_t));
            array224->size = 137;
            array224->refC = 1;
            array224->id = 224;
            array224->data = (unsigned int*)malloc(array224->size*sizeof(unsigned int));
            memset(array224->data, 0, array224->size*sizeof(unsigned int));
            DEBUG_NEW(array224->id);
         }
         unsigned int loop66 = 0;
         unsigned int loopLimit66 = (rand()%loopsFactor)/3 + 1;
         for(; loop66 < loopLimit66; loop66++) {
            for (int i = 0; i < array224->size; i++) {
               array224->data[i]--;
            }
            for (int i = 0; i < array224->size; i++) {
               if (array224->data[i] == 43) { 
                  return array224;
               }
            }
         }
         array_t_param params0;
         params0.size = 1;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         params0.data[0] = array224;
         array_t* array225 = func24(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array225->id);
         free(params0.data);
         array225->refC--;
         if(array225->refC == 0) {
            free(array225->data);
            free(array225);
            DEBUG_FREE(array225->id);
         }
         array224->refC--;
         if(array224->refC == 0) {
            free(array224->data);
            free(array224);
            DEBUG_FREE(array224->id);
         }
      }
      else {
         array_t_param params0;
         params0.size = 0;
         params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
         array_t* array226 = func16(&params0, rng(), loopsFactor);
         DEBUG_RETURN(array226->id);
         free(params0.data);
         array226->refC--;
         if(array226->refC == 0) {
            free(array226->data);
            free(array226);
            DEBUG_FREE(array226->id);
         }
      }
      array_t* array227;
      if (pCounter > 0) {
         array227 = vars->data[--pCounter];
         array227->refC++;
         DEBUG_COPY(array227->id);
      } else {
         array227 = (array_t*)malloc(sizeof(array_t));
         array227->size = 710;
         array227->refC = 1;
         array227->id = 227;
         array227->data = (unsigned int*)malloc(array227->size*sizeof(unsigned int));
         memset(array227->data, 0, array227->size*sizeof(unsigned int));
         DEBUG_NEW(array227->id);
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array227;
      array_t* array228 = func17(&params0, loopsFactor);
      DEBUG_RETURN(array228->id);
      free(params0.data);
      if(PATH0 & 2) {
         array_t* array229;
         if (pCounter > 0) {
            array229 = vars->data[--pCounter];
            array229->refC++;
            DEBUG_COPY(array229->id);
         } else {
            array229 = (array_t*)malloc(sizeof(array_t));
            array229->size = 836;
            array229->refC = 1;
            array229->id = 229;
            array229->data = (unsigned int*)malloc(array229->size*sizeof(unsigned int));
            memset(array229->data, 0, array229->size*sizeof(unsigned int));
            DEBUG_NEW(array229->id);
         }
         unsigned int loop67 = 0;
         unsigned int loopLimit67 = (rand()%loopsFactor)/3 + 1;
         for(; loop67 < loopLimit67; loop67++) {
            for (int i = 0; i < array229->size; i++) {
               array229->data[i]--;
            }
            for (int i = 0; i < array229->size; i++) {
               if (array229->data[i] == 90) { 
                  return array229;
               }
            }
         }
         array_t_param params1;
         params1.size = 3;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array227;
         params1.data[1] = array228;
         params1.data[2] = array229;
         array_t* array230 = func28(&params1, rng(), loopsFactor);
         DEBUG_RETURN(array230->id);
         free(params1.data);
         array230->refC--;
         if(array230->refC == 0) {
            free(array230->data);
            free(array230);
            DEBUG_FREE(array230->id);
         }
         array229->refC--;
         if(array229->refC == 0) {
            free(array229->data);
            free(array229);
            DEBUG_FREE(array229->id);
         }
      }
      else {
         array_t_param params1;
         params1.size = 2;
         params1.data = (array_t**)malloc(params1.size*sizeof(array_t*));
         params1.data[0] = array227;
         params1.data[1] = array228;
         array_t* array231 = func22(&params1, rng(), loopsFactor);
         DEBUG_RETURN(array231->id);
         free(params1.data);
         array231->refC--;
         if(array231->refC == 0) {
            free(array231->data);
            free(array231);
            DEBUG_FREE(array231->id);
         }
      }
      for (int i = 0; i < array228->size; i++) {
         if (array228->data[i] == 33) { 
            return array228;
         }
      }
      array228->refC--;
      if(array228->refC == 0) {
         free(array228->data);
         free(array228);
         DEBUG_FREE(array228->id);
      }
      array227->refC--;
      if(array227->refC == 0) {
         free(array227->data);
         free(array227);
         DEBUG_FREE(array227->id);
      }
   }
   array_t* array232;
   if (pCounter > 0) {
      array232 = vars->data[--pCounter];
      array232->refC++;
      DEBUG_COPY(array232->id);
   } else {
      array232 = (array_t*)malloc(sizeof(array_t));
      array232->size = 685;
      array232->refC = 1;
      array232->id = 232;
      array232->data = (unsigned int*)malloc(array232->size*sizeof(unsigned int));
      memset(array232->data, 0, array232->size*sizeof(unsigned int));
      DEBUG_NEW(array232->id);
   }
   return array232;
}

File: data/multilang_comparison/program_julia/program_julia/src/func7.jl
include("program_julia_head.jl")
function func7(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   loopLimit56 = (50)/2 + 1
   for i in 1:loopLimit56 
      if (PATH0 & 1) != 0
         array192 = nothing
         if pCounter > 1
             array192 = vars.data[pCounter]
             pCounter -= 1
             array192.refC += 1
         else
             array192 = JArray(zeros(Int, 462), 462, 1, 192)
         end
         loopLimit57 = (50)/3 + 1
         for i in 1:loopLimit57 
            for i in 0:(array192.size-1) 
               array192.data[i+1] -= 1
            end
            for i in 1:array192.size
                if array192.data[i] == 92
                        return array192
               end
            end
         end
         params0 = JArray_param{Int}()
         params0.size = 1
         push!(params0.data, array192)
         params0.size += 1
         array193 = func24(params0, get_path(), loopsFactor);
         array193.refC -= 1
         if array193.refC == 0
             empty!(array193.data)
             array193 = nothing
         end
         array192.refC -= 1
         if array192.refC == 0
             empty!(array192.data)
             array192 = nothing
         end
      
      else 
         params0 = JArray_param{Int}()
         params0.size = 0
         array194 = func16(params0, get_path(), loopsFactor);
         array194.refC -= 1
         if array194.refC == 0
             empty!(array194.data)
             array194 = nothing
         end
      end
      array195 = nothing
      if pCounter > 1
          array195 = vars.data[pCounter]
          pCounter -= 1
          array195.refC += 1
      else
          array195 = JArray(zeros(Int, 157), 157, 1, 195)
      end
      params0 = JArray_param{Int}()
      params0.size = 1
      push!(params0.data, array195)
      params0.size += 1
      array196 = func17(params0, loopsFactor);
      if (PATH0 & 2) != 0
         array197 = nothing
         if pCounter > 1
             array197 = vars.data[pCounter]
             pCounter -= 1
             array197.refC += 1
         else
             array197 = JArray(zeros(Int, 947), 947, 1, 197)
         end
         loopLimit58 = (50)/3 + 1
         for i in 1:loopLimit58 
            for i in 0:(array196.size-1) 
               array196.data[i+1] -= 1
            end
            for i in 1:array195.size
                if array195.data[i] == 21
                        return array195
               end
            end
         end
         params1 = JArray_param{Int}()
         params1.size = 3
         push!(params1.data, array195)
         params1.size += 1
         push!(params1.data, array196)
         params1.size += 1
         push!(params1.data, array197)
         params1.size += 1
         array198 = func28(params1, get_path(), loopsFactor);
         array198.refC -= 1
         if array198.refC == 0
             empty!(array198.data)
             array198 = nothing
         end
         array197.refC -= 1
         if array197.refC == 0
             empty!(array197.data)
             array197 = nothing
         end
      
      else 
         params1 = JArray_param{Int}()
         params1.size = 2
         push!(params1.data, array195)
         params1.size += 1
         push!(params1.data, array196)
         params1.size += 1
         array199 = func22(params1, get_path(), loopsFactor);
         array199.refC -= 1
         if array199.refC == 0
             empty!(array199.data)
             array199 = nothing
         end
      end
      for i in 1:array195.size
          if array195.data[i] == 25
                  return array195
         end
      end
      array196.refC -= 1
      if array196.refC == 0
          empty!(array196.data)
          array196 = nothing
      end
      array195.refC -= 1
      if array195.refC == 0
          empty!(array195.data)
          array195 = nothing
      end
   end
   array200 = nothing
   if pCounter > 1
       array200 = vars.data[pCounter]
       pCounter -= 1
       array200.refC += 1
   else
       array200 = JArray(zeros(Int, 81), 81, 1, 200)
   end
   return array200;
end

